### PR TITLE
Make the AI gap analysis comprehensive

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
Previously, the gap analysis response listed a summary of what tests might be missing to achieve minimum satisfactory test coverage, as well as having a "Prioritized Recommendations" section for giving suggestions on some tests to add to increase test coverage. However, this is not sufficient, and ideally the gap analysis would identify every single test that is required to meet the minimum bar in test coverage.

This PR updates the gap analysis prompt to adjust the structure of the response to include every test that will be required to meet the minimum bar. If a feature owner adds all the suggested tests, it should be assumed that the subsequent gap analysis will pass with no suggestions.

### CODE REVIEW

File: templates/prompts/gap-analysis.html [hunk 1]
```diff
@@ -86,16 +88,20 @@
...
 
 -----
 
-### Prioritized Recommendation
+### Required Tests to Achieve Minimum Satisfactory Coverage
+
+This is the most critical section. Based on the "Gaps & Notes" identified above, provide a checklist of every new test that must be written.
 
-Based on the gaps, provide a 1-3 bullet-point list of the *most critical* tests that must be written to achieve a minimum satisfactory level of coverage.
+**Format:** Group the tests by the 5 definition categories. If a category has no gaps, omit it.
```
Comment (jcscottiii) [inline templates/prompts/gap-analysis.html]: Do you want to print the category but say "no gaps" in that case?

I'm wondering about the case of the API owner reviewing the AI report for the first time. Would they know that a category was evaluated if it was omitted?

**Review by jcscottiii — APPROVED** [review]
LGTM with one question.

File: templates/prompts/gap-analysis.html (hunk 1 above)
Comment (DanielRyanSmith) [reply templates/prompts/gap-analysis.html]: That's a good idea - I've reworded this to keep the category name with a default of "No gaps." when no test suggestions exist. 🙂
